File: Data Structures & Algorithms/maximum-nesting-depth-of-the-parentheses/submission-0.py
from collections import deque
class Solution:
    def maxDepth(self, s: str) -> int:
        stack = deque()
        ans = 0
        curSize =0 
        for char in s:
            if char == '(':
                stack.append(char)
                curSize+=1
                ans = max(ans,curSize)
            elif (char ==')' and len(stack)!=0):
                    stack.pop()
                    curSize -=1
        return ans

File: Data Structures & Algorithms/daily-temperatures/submission-0.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int>ans(temperatures.size(),0);
        stack<pair<int,int>>stk; //value,index
        for(int i = 0; i<temperatures.size();i++){
            int curVal = temperatures[i];
            while(!stk.empty() && curVal>stk.top().first){
                auto pair = stk.top();
                stk.pop();
                ans[pair.second] = i - pair.second;
            }
            stk.push({curVal,i});
        }
        return ans;
    }
};
